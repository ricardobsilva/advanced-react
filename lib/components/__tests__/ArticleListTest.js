import React from 'react';

import ArticleList from '../ArticleList';
import Article from '../Article';
//import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import 'babel-polyfill'

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: {id: 'abc'},
      b: {id: 'abcde'},
    },
  };

  Article.propTypes = {};

  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );
    expect(wrapper.find('Article').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
