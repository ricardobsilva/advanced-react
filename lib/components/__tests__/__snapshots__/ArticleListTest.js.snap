// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "a": Object {
          "id": "abc",
        },
        "b": Object {
          "id": "abcde",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Article
          article={
            Object {
              "id": "abc",
            }
          }
        />,
        <Article
          article={
            Object {
              "id": "abcde",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "abc",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "id": "abc",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "abcde",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "id": "abcde",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Article
            article={
              Object {
                "id": "abc",
              }
            }
          />,
          <Article
            article={
              Object {
                "id": "abcde",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "abc",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "id": "abc",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "abcde",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "id": "abcde",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
